packages/samples/specs/resource-manager/legacy/legacy-operations/main.tsp

```diff
@@ -0,0 +1,108 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using Http;
+using Rest;
+using Versioning;
+using Azure.ResourceManager;
+
+/** Contoso Resource Provider management API. */
+#suppress "@azure-tools/typespec-azure-resource-manager/missing-operations-endpoint" ""
+@armProviderNamespace
+@service(#{ title: "ContosoProviderHubClient" })
+@versioned(Versions)
+namespace Microsoft.ContosoProviderHub;
+
+/** Contoso API versions */
+enum Versions {
+  /** 2021-10-01-preview version */
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  @armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v5)
+  `2021-10-01-preview`,
+}
+
+/** Standard operation list */
+interface Operations extends Azure.ResourceManager.Operations {}
+
+/** A best practice resource */
+model BestPractice is Azure.ResourceManager.ProxyResource<BestPracticeProperties> {
+  ...ResourceNameParameter<
+    Resource = BestPractice,
+    KeyName = "bestPracticeName",
+    SegmentName = "bestPractices",
+    NamePattern = ""
+  >;
+}
+
+/** Resource-specific properties of a best practice resource */
+model BestPracticeProperties {
+  ...DefaultProvisioningStateProperty;
+
+  /** The description of the best practice */
+  description: string;
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-interface-requires-decorator" "Resource template"
```

Comment:
does this actually emit a warning here?
Seems like we should file an issue if it does